runelite-client/src/main/java/net/runelite/client/plugins/puzzlesolver/solver/pathfinding/IDAStarMM.java

```diff
@@ -0,0 +1,787 @@
+
+	private void moveTowardsVal(int valTarget, int x, int y, boolean rowMode)
+	{
+		//Not in place
+		boolean reached = false;
+
+		int row = y;
+		int column = x;
+
+		while (currentState.getPiece(x, y) != valTarget)
+		{
+			//Find piece location
+			Point locVal = findPiece(valTarget);
+			Point locBlank = findPiece(BLANK_TILE_VALUE);
+
+			if (!reached)
+			{
+				int distX = locVal.getX() - locBlank.getX();
+				int distY = locVal.getY() - locBlank.getY();
+				int distAbsX = Math.abs(distX);
+				int distAbsY = Math.abs(distY);
+
+				if (distX == 0)
+				{
+					//Same column
+					if (distAbsY == 1)
+					{
+						//Next to
+						reached = true;
+					}
+					else
+					{
+						//More than 2 away, move towards on Y-axis
+						if (distY >= 2)
+						{
+							Point locSwap = new Point(locBlank.getX(), locBlank.getY() + 1);
+							swap(locBlank.getX(), locBlank.getY(), locSwap.getX(), locSwap.getY());
+						}
+						else if (distY <= -2)
+						{
+							Point locSwap = new Point(locBlank.getX(), locBlank.getY() - 1);
+							swap(locBlank.getX(), locBlank.getY(), locSwap.getX(), locSwap.getY());
+						}
+					}
+				}
+				else if (distY == 0)
+				{
+					//Same row
+					if (distAbsX == 1)
+					{
+						//Next to
+						reached = true;
+					}
+					else
+					{
+						//More than 2 away, move towards on X-axis
+						if (distX >= 2)
+						{
+							Point locSwap = new Point(locBlank.getX() + 1, locBlank.getY());
+							swap(locBlank.getX(), locBlank.getY(), locSwap.getX(), locSwap.getY());
+						}
+						else if (distX <= -2)
+						{
+							Point locSwap = new Point(locBlank.getX() - 1, locBlank.getY());
+							swap(locBlank.getX(), locBlank.getY(), locSwap.getX(), locSwap.getY());
+						}
+					}
+				}
+				else
+				{
+					//Different row and column
+					if (rowMode)
+					{
+						//Check if already correct above
+						if (locBlank.getY() - 1 == row &&
+							validRowNumbers.get(row).contains(currentState.getPiece(locBlank.getX(), locBlank.getY() - 1)) &&
+							currentState.getPiece(locBlank.getX(), locBlank.getY() - 1) < valTarget &&
+							distY <= -1)
+						{
+							//Move forward
+							Point locSwap = new Point(locBlank.getX() + 1, locBlank.getY());
+							swap(locBlank.getX(), locBlank.getY(), locSwap.getX(), locSwap.getY());
+							continue;
+						}
+
+						//Move upwards or downwards
+						if (distY >= 1)
+						{
+							Point locSwap = new Point(locBlank.getX(), locBlank.getY() + 1);
+							swap(locBlank.getX(), locBlank.getY(), locSwap.getX(), locSwap.getY());
+						}
+						else if (distY <= -1)
+						{
+							Point locSwap = new Point(locBlank.getX(), locBlank.getY() - 1);
+							swap(locBlank.getX(), locBlank.getY(), locSwap.getX(), locSwap.getY());
+						}
+					}
+					else
+					{
+						//Check if already correct to the left
+						if (locBlank.getX() - 1 == column &&
+							validColumnNumbers.get(column).contains(currentState.getPiece(locBlank.getX() - 1, locBlank.getY())) &&
```

Comment:
The ``&&`` should come before the condition. Look at for example https://github.com/runelite/runelite/blob/master/runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksPlugin.java#L113.

Reply:
I think i spaced it one tab to much?

Reply:
Yeah it should only be 1 tab.